### ISSUE TITLE
"make" Commend Error

### ISSUE THREAD

**Issue by acv-thanld** [pjreddie/darknet#535]
make
gcc -Iinclude/ -Isrc/ -DOPENCV `pkg-config --cflags opencv`  -DGPU -I/usr/local/cuda/include/ -DCUDNN  -Wall -Wno-unknown-pragmas -Wfatal-errors -fPIC -O0 -g -DOPENCV -DGPU -DCUDNN -c ./src/gemm.c -o obj/gemm.o
In file included from ./src/utils.h:5:0,
                 from ./src/gemm.c:2:
include/darknet.h:19:14: fatal error: cudnn.h: No such file or directory
     #include "cudnn.h"
              ^~~~~~~~~
compilation terminated.
Makefile:85: recipe for target 'obj/gemm.o' failed
make: *** [obj/gemm.o] Error 1

**Comment by ahsan856jalal** [comment]
can you show use the configuration of the Makefile ? 
in the makefile, try to make with GPU=0, CudNN=0, Opencv=0, Openmp=0, Debug=0

**Comment by liangxi627** [comment]
How did you solved the problem? @acv-thanld , I meet the following error, @ahsan856jalal please help me, thank you!

> In file included from ./src/cuda.h:4:0,
                 from ./src/cuda.c:5:
include/darknet.h:19:23: fatal error: cudnn.h: No such file or directory
compilation terminated.
In file included from ./src/activations.h:3:0,
                 from ./src/activations.c:1:
include/darknet.h:19:23: fatal error: cudnn.h: No such file or directory
compilation terminated.
Makefile:85: recipe for target 'obj/cuda.o' failed
make: *** [obj/cuda.o] Error 1
make: *** Waiting for unfinished jobs....
Makefile:85: recipe for target 'obj/activations.o' failed
make: *** [obj/activations.o] Error 1
In file included from ./src/list.h:3:0,
                 from ./src/list.c:3:
include/darknet.h:19:23: fatal error: cudnn.h: No such file or directory
compilation terminated.
Makefile:85: recipe for target 'obj/list.o' failed
make: *** [obj/list.o] Error 1
In file included from ./src/cuda.h:4:0,
                 from ./src/convolutional_layer.h:4,
                 from ./src/convolutional_layer.c:1:
include/darknet.h:19:23: fatal error: cudnn.h: No such file or directory
compilation terminated.
Makefile:85: recipe for target 'obj/convolutional_layer.o' failed
make: *** [obj/convolutional_layer.o] Error 1
In file included from ./src/utils.h:5:0,
                 from ./src/gemm.c:2:
include/darknet.h:19:23: fatal error: cudnn.h: No such file or directory
compilation terminated.
Makefile:85: recipe for target 'obj/gemm.o' failed
make: *** [obj/gemm.o] Error 1
In file included from ./src/utils.h:5:0,
                 from ./src/utils.c:11:
include/darknet.h:19:23: fatal error: cudnn.h: No such file or directory
compilation terminated.
Makefile:85: recipe for target 'obj/utils.o' failed
make: *** [obj/utils.o] Error 1
In file included from ./src/box.h:3:0,
                 from ./src/image.h:9,
                 from ./src/image.c:1:
include/darknet.h:19:23: fatal error: cudnn.h: No such file or directory
compilation terminated.
Makefile:85: recipe for target 'obj/image.o' failed
make: *** [obj/image.o] Error 1
In file included from ./src/cuda.h:4:0,
                 from ./src/deconvolutional_layer.h:4,
                 from ./src/deconvolutional_layer.c:1:
include/darknet.h:19:23: fatal error: cudnn.h: No such file or directory
compilation terminated.
Makefile:85: recipe for target 'obj/deconvolutional_layer.o' failed
make: *** [obj/deconvolutional_layer.o] Error 1


My makefile is like this:

> GPU=1
CUDNN=1
OPENCV=1
OPENMP=0
DEBUG=0
NVCC=/A-pool/cuda/cuda-8.0/bin/nvcc
ARCH= -gencode arch=compute_30,code=sm_30 \
      -gencode arch=compute_35,code=sm_35 \
      -gencode arch=compute_50,code=[sm_50,compute_50] \
      -gencode arch=compute_52,code=[sm_52,compute_52] \
      -gencode arch=compute_61,code=[sm_61,compute_61] 

# This is what I use, uncomment if you know your arch and want to specify
# ARCH= -gencode arch=compute_52,code=compute_52

VPATH=./src/:./examples
SLIB=libdarknet.so
ALIB=libdarknet.a
EXEC=darknet
OBJDIR=./obj/

CC=gcc
NVCC=nvcc 
AR=ar
ARFLAGS=rcs
OPTS=-Ofast
LDFLAGS= -lm -pthread 
COMMON= -Iinclude/ -Isrc/
CFLAGS=-Wall -Wno-unused-result -Wno-unknown-pragmas -Wfatal-errors -fPIC

ifeq ($(OPENMP), 1) 
CFLAGS+= -fopenmp
endif

ifeq ($(DEBUG), 1) 
OPTS=-O0 -g
endif

CFLAGS+=$(OPTS)

ifeq ($(OPENCV), 1) 
COMMON+= -DOPENCV
CFLAGS+= -DOPENCV
LDFLAGS+= `pkg-config --libs opencv` 
COMMON+= `pkg-config --cflags opencv` 
endif

ifeq ($(GPU), 1) 
COMMON+= -DGPU -I/A-pool/cuda/cuda-8.0/include/
CFLAGS+= -DGPU
LDFLAGS+= -L/A-pool/cuda/cuda-8.0/lib64 -lcuda -lcudart -lcublas -lcurand
endif

ifeq ($(CUDNN), 1) 
COMMON+= -DCUDNN 
CFLAGS+= -DCUDNN
LDFLAGS+= -lcudnn
endif

OBJ=gemm.o utils.o cuda.o deconvolutional_layer.o convolutional_layer.o list.o image.o activations.o im2col.o col2im.o blas.o crop_layer.o dropout_layer.o maxpool_layer.o softmax_layer.o data.o matrix.o network.o connected_layer.o cost_layer.o parser.o option_list.o detection_layer.o route_layer.o upsample_layer.o box.o normalization_layer.o avgpool_layer.o layer.o local_layer.o shortcut_layer.o logistic_layer.o activation_layer.o rnn_layer.o gru_layer.o crnn_layer.o demo.o batchnorm_layer.o region_layer.o reorg_layer.o tree.o  lstm_layer.o l2norm_layer.o yolo_layer.o
EXECOBJA=captcha.o lsd.o super.o art.o tag.o cifar.o go.o rnn.o segmenter.o regressor.o classifier.o coco.o yolo.o detector.o nightmare.o darknet.o
ifeq ($(GPU), 1) 
LDFLAGS+= -lstdc++ 
OBJ+=convolutional_kernels.o deconvolutional_kernels.o activation_kernels.o im2col_kernels.o col2im_kernels.o blas_kernels.o crop_layer_kernels.o dropout_layer_kernels.o maxpool_layer_kernels.o avgpool_layer_kernels.o
endif

EXECOBJ = $(addprefix $(OBJDIR), $(EXECOBJA))
OBJS = $(addprefix $(OBJDIR), $(OBJ))
DEPS = $(wildcard src/*.h) Makefile include/darknet.h

#all: obj backup results $(SLIB) $(ALIB) $(EXEC)
all: obj  results $(SLIB) $(ALIB) $(EXEC)


$(EXEC): $(EXECOBJ) $(ALIB)
	$(CC) $(COMMON) $(CFLAGS) $^ -o $@ $(LDFLAGS) $(ALIB)

$(ALIB): $(OBJS)
	$(AR) $(ARFLAGS) $@ $^

$(SLIB): $(OBJS)
	$(CC) $(CFLAGS) -shared $^ -o $@ $(LDFLAGS)

$(OBJDIR)%.o: %.c $(DEPS)
	$(CC) $(COMMON) $(CFLAGS) -c $< -o $@

$(OBJDIR)%.o: %.cu $(DEPS)
	$(NVCC) $(ARCH) $(COMMON) --compiler-options "$(CFLAGS)" -c $< -o $@

obj:
	mkdir -p obj
backup:
	mkdir -p backup
results:
	mkdir -p results

.PHONY: clean

clean:
	rm -rf $(OBJS) $(SLIB) $(ALIB) $(EXEC) $(EXECOBJ) $(OBJDIR)/*
